cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -435,16 +436,59 @@ private static <I extends ReactiveLimitsHolder & Injection<I>> void writeSynchro
         SynchronousMachineEq.write(context.getNamingStrategy().getCgmesId(i), i.getNameOrId(),
                 context.getNamingStrategy().getCgmesId(i.getTerminal().getVoltageLevel()),
                 generatingUnit, regulatingControlId, reactiveLimitsId, minQ, maxQ, ratedS, defaultRatedS, kind, cimNamespace, writer, context);
+
         if (generatingUnit != null && !generatingUnitsWritten.contains(generatingUnit)) {
+
+            String hydroPowerPlantId = generatingUnitWriteHydroPowerPlantAndFossilFuel(i, cimNamespace, energySource, generatingUnit, writer, context);
+
             // We have not preserved the names of generating units
             // We name generating units based on the first machine found
             String generatingUnitName = "GU_" + i.getNameOrId();
+
             GeneratingUnitEq.write(generatingUnit, generatingUnitName, energySource, minP, maxP, targetP, cimNamespace, writeInitialP,
-                    i.getTerminal().getVoltageLevel().getSubstation().map(s -> context.getNamingStrategy().getCgmesId(s)).orElse(null), writer, context);
+                    i.getTerminal().getVoltageLevel().getSubstation().map(s -> context.getNamingStrategy().getCgmesId(s)).orElse(null),
+                    hydroPowerPlantId, writer, context);
             generatingUnitsWritten.add(generatingUnit);
         }
     }
 
+    private static <I extends ReactiveLimitsHolder & Injection<I>> String generatingUnitWriteHydroPowerPlantAndFossilFuel(I i, String cimNamespace, EnergySource energySource, String generatingUnit, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
+        String hydroPlantStorageType = i.getProperty(Conversion.PROPERTY_CGMES_SYNCHRONOUS_MACHINE_HYDRO_PLANT_STORAGE_KIND);
+        String hydroPowerPlantId = null;
+        if (hydroPlantStorageType != null && GeneratingUnitEq.generatingUnitClassName(energySource).equals("HydroGeneratingUnit")) {
```

Comment:
You don't need to make the generatingUnitClassName method public.
To verify the type of unit, you can directly check the value of energySource:
```if (hydroPlantStorageType != null && energySource.equals(EnergySource.HYDRO))```

Reply:
Done